frontend/components/Documents.js

```diff
@@ -43,6 +110,35 @@ const Documents = () => {
             <p>
                 This page displays all the documents stored in backend.
             </p>
+            <div>
+                <form onSubmit={handleSubmit}>
+                    <div className="form-group">
+                        <label htmlFor="author">Author</label>
+                        <input type="text" className="form-control"
+                            id="author" value={newDocData.author}
+                            onChange={handleAuthorInputChange}/>
+                    </div>
+                    <div className="form-group">
+                        <label htmlFor="title">Title</label>
+                        <input type="text" className="form-control"
+                            id="title" value={newDocData.title}
+                            onChange={handleTitleInputChange}/>
+                    </div>
+                    <div className="form-group">
+                        <label htmlFor="date">Date</label>
+                        <input type="number" className="form-control"
+                            id="date" value={newDocData.date}
+                            onChange={handleDateInputChange}/>
+                    </div>
+                    <div className="form-group">
+                        <label htmlFor="text">Text</label>
+                        <textarea className="form-control" id="text"
+                            rows="8" value={newDocData.text}
+                            onChange={handleTextInputChange}></textarea>
+                    </div>
```

Comment:
Another opinionated HTML comment:

This pattern here, of using a `<div>` to wrap a `<label>` and associated input element, could also be composed as:
```
<label>
  <p>Text</p>
  <input />
</label>
```

In other words, remove the semantically meaningless `<div>`, wrap a text block and an input element in the `<label>` element and don't worry about including the explicit `for=` attributes. The wrapping `<label>` will semantically associate text content and input element as expected.

Reply:
The form has been updated.